syslogparser.go

```diff
@@ -32,18 +38,18 @@ func DetectRFC(buff []byte) (RFC, error) {
 	var v int
 	var err error
 
-	for i := 0; i < max; i++ {
-		if buff[i] == '>' && i < max {
-			x := i + 1
-
-			v, err = parsercommon.ParseVersion(
-				buff, &x, max,
-			)
+	if max > len(buff) {
+		max = len(buff)
+	}
 
-			break
-		}
+	idx := bytes.IndexByte(buff, '>')
+	if idx == -1 || idx >= max {
+		//there is no complete header for RFC5424, throw RFC_3164
+		return RFC_3164, nil
```

Comment:
3164 also requires a <> portion before the header for the PRI field. This doesn't look right.